libraries/AP_Scripting/generator/src/main.c

```diff
@@ -901,15 +936,20 @@ void handle_ap_object(void) {
     node->flags |= UD_FLAG_SEMAPHORE;
   } else if (strcmp(type, keyword_scheduler_semaphore) == 0) {
     node->flags |= UD_FLAG_SCHEDULER_SEMAPHORE;
+  } else if (strcmp(type, keyword_semaphore_pointer) == 0) {
+    node->flags |= UD_FLAG_SEMAPHORE_POINTER;
   } else if (strcmp(type, keyword_method) == 0) {
     handle_method(node->name, &(node->methods));
   } else {
     error(ERROR_SINGLETON, "AP_Objects only support aliases, methods or semaphore keyowrds (got %s)", type);
   }
 
   // check that we didn't just add 2 singleton flags
-  if ((node->flags & UD_FLAG_SEMAPHORE) && (node->flags & UD_FLAG_SCHEDULER_SEMAPHORE)) {
-    error(ERROR_SINGLETON, "Taking both a library semaphore and scheduler semaphore is prohibited");
+  int SEMAPHORE = (node->flags & UD_FLAG_SEMAPHORE)?1:0;
```

Comment:
lower case? I'm just not sure why this is caps locked.

Reply:
Its almost as though I copied and passed it from the enum..... fixed